hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/BucketEndpoint.java

```diff
@@ -341,7 +369,13 @@ public Response head(@PathParam("bucket") String bucketName)
       throws OS3Exception, IOException {
     S3GAction s3GAction = S3GAction.HEAD_BUCKET;
     try {
-      getBucket(bucketName);
+      captureLatencyNs(getLatencyMetrics().getHeadBucketLatencyNs(), () -> {
+        try {
+          getBucket(bucketName);
+        } catch (OS3Exception ex) {
+          throw new RuntimeException(ex);
+        }
```

Comment:
@tanvipenumudy acceptance test is failing with:

```
Head Bucket not existent                                              | FAIL |
'
An error occurred (500) when calling the HeadBucket operation (reached max retries: 4): Internal Server Error' does not contain '404'
```

https://github.com/tanvipenumudy/ozone/actions/runs/4571528625/jobs/8070135759#step:5:475
https://github.com/tanvipenumudy/ozone/actions/runs/4571528625/jobs/8070135913#step:5:337
https://github.com/tanvipenumudy/ozone/actions/runs/4571528625/jobs/8070135972#step:5:249
https://github.com/tanvipenumudy/ozone/actions/runs/4571528625/jobs/8070136118#step:5:708

I guess it happens because of this exception translation.  The wrapped `OS3Exception` should be rethrown in the outer `catch`.

Reply:
Thank you @adoroszlai for the review, made the required changes.